src/memote/metabolites.jl

```diff
@@ -0,0 +1,120 @@
+#=
+This file contains a collection of tests based on Memote. See Lieven, C., Beber,
+M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
+testing. Nat Biotechnol 38, 272–276 (2020).
+https://doi.org/10.1038/s41587-020-0446-y for details.
+=#
+
+"""
+$(TYPEDSIGNATURES)
+
+Return a list of all boundary reactions that allow flux into the model and
+create a metabolite. Assume that boundary reactions only create a single
+metabolite. Set the kwarg `only_into=false` to also return metabolites that can
+be produced by the model under default conditions.
+"""
+function metabolites_medium_components(model; only_into=true)
+    mets = String[]
+    for (rid, lb, ub) in zip(reactions(model), bounds(model)...)
+        if is_boundary(model, rid)
+            mid = first(keys(reaction_stoichiometry(model, rid))) 
+            only_into && lb < 0 && ub <= 0 && push!(mets, mid)
+            !only_into && lb < 0 && push!(mets, mid)
+        end
+    end
+    mets
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+List all metabolites without a formula. Use `corner_cases` to specify an extra
+case to check for formula's that are not properly assigned.
+"""
+metabolites_no_formula(model; corner_cases = ["X", "x", ""]) = [mid for mid in metabolites(model) if isnothing(metabolite_formula(model, mid)) || metabolite_formula(model, mid) in corner_cases]
```

Comment:
`corner_cases` are not corner cases, call it `ignore_values` or so. Also pls do not default the parameter here; either make it a global constant or force the user to specify this in a front-end function (which may have centralized defaults)

Reply:
centralized defaults, see latest commit